### ISSUE TITLE
Error when rendering A-- This is the text -- B

### ISSUE THREAD

**Issue by npatmaja** [knsv/mermaid#102]
I am experiencing an error while parsing *text on link* (copy pasted from [flowchart basic syntax](http://knsv.github.io/mermaid/flowchart.html)) using [mermaid 0.3.2](https://cdn.rawgit.com/knsv/mermaid/0.3.2/dist/mermaid.full.min.js)
```
A-- This is the text -- B
```
the error is:
```
Uncaught Error: Parse error on line 2:
...his is the text -- B
-----------------------^
Expecting 'GRAPH', 'SPACE', 'DIR', 'TAGEND', 'TAGSTART', 'subgraph', 'end', 'MINUS', '--', '==', 'STYLE', 'LINKSTYLE', 'CLASSDEF', 'CLASS', 'CLICK', 'NUM', 'COMMA', 'ALPHA', 'COLON', 'BRKT', 'DOT', 'PLUS', 'EQUALS', 'MULT', got 'EOF'
```
However, mermaid renders perfectly `A---|This is the text|B;` syntax. 
Is `-- This is the text --` no longer supported? if so please change the documentation.

**Comment by knsv** [comment]
I see the problem. There were no test case for that syntax. Will look at it.

**Comment by knsv** [comment]
So text can contain -- and it can contain space. So the ending element " --" is hard for the parser to separate from text.

I can see that having things like 
```
count--
```
in the boxes makes sense.

Right now it works --- as ending. Perhaps thats the way to go or perhaps forbidding -- in text is the best option. I am open for opinions.

**Comment by npatmaja** [comment]
I think having `---` as ending will do as the other lines with text also use three characters as ending, e.g., `.->` or `==>`. 
Another alternative is to have this kind of syntax:
```
A--|some long text that uses escaped char like `||--B
```

**Comment by knsv** [comment]
Then think I will update the docs showing the ```---```as ending!

**Comment by npatmaja** [comment]
Cool! thanks. I'll download the latest library :)

**Comment by knsv** [comment]
I think this one should be closed as of 0.3.3. @npatmaja, I saw your guide for using mermaid with docpad, nice work!
